core/math/aabb.cpp

```diff
@@ -123,10 +123,9 @@ bool AABB::intersects_ray(const Vector3 &p_from, const Vector3 &p_dir, Vector3 *
 		ERR_PRINT("AABB size is negative, this is not supported. Use AABB.abs() to get an AABB with a positive size.");
```

Comment:
Name of parameter `r_clip` should be changed to e.g. `r_intersection_point`, if it no longer returns a "clip" (whatever that is).

Reply:
I'm mostly trying to just get it to match the other functions like the intersects_segment code which is directly below that.
This was mostly a fix rather than a rewrite.

It seems that the seg func is used all over and uses this convention.

Reply:
> It seems that the seg func is used all over and uses this convention.

This is a logical fallacy - this doesn't mean it's correct, and in this case, you have _changed what the function does_ so it now returns an intersection point, so the parameter name should be changed to reflect this.

Reply:
It's not a "logical fallacy" its trying to minimize the diff and remain consistent with a code base I am unfamiliar with.
r_clip is clearly the format of return clipping point which given axis aligned bounding boxes are typically used for clipping operations makes perfect sense.

To move it along and keep the two functions consistent I've renamed to r_intersection_point as it seems the convention r_clip isn't used outside this file